src/libPMacc/include/cuSTL/algorithm/mpi/Gather.tpp

```diff
@@ -31,6 +31,41 @@ namespace algorithm
 {
 namespace mpi
 {
+    
+namespace GatherHelper
+{
+    
+template<int dim, int memDim>
+struct posInMem;
+
+template<int dim>
+struct posInMem<dim, dim>
+{
+    math::Int<dim> operator()(const math::Int<dim>& pos, int) const
```

Comment:
I am not sure that works for 4D and above ;) pls write explicit implementation for 1D, too and do not implement the general case, so it throws an error.